src/Provider/RewriteRules.php

```diff
@@ -0,0 +1,115 @@
+<?php
+ /**
+  * Register rewrite rules.
+  *
+  * @package SatisPress
+  * @license GPL-2.0-or-later
+  * @since 0.3.0
+  */
+
+declare ( strict_types = 1 );
+
+namespace SatisPress\Provider;
+
+use Cedaro\WP\Plugin\AbstractHookProvider;
+
+/**
+ * Class to register rewrite rules.
+ *
+ * @since 0.3.0
+ */
+class RewriteRules extends AbstractHookProvider {
+	/**
+	 * Register hooks.
+	 *
+	 * @since 0.3.0
+	 */
+	public function register_hooks() {
+		add_filter( 'query_vars',                              array( $this, 'register_query_vars' ) );
+		add_action( 'init',                                    array( $this, 'register_rewrite_rules' ) );
+		add_action( 'generate_rewrite_rules',                  array( $this, 'register_external_rewrite_rules' ) );
+		register_activation_hook( $this->plugin->get_file(),   array( $this, 'activate' ) );
```

Comment:
If the activation / deactivation routine included anything other than updating / clearing rewrite rules, then it wouldn't make sense to keep this here.

I'd suggest moving these to their own provider classes (one for activation, one for deactivation).

Reply:
If another other activation/deactivation routines are needed, these could always be moved out or separate routines could be registered. I typically include a `Setup` or `Install` provider if there's a more involved activation process.

I think keeping them here adds a bit of clarity to how rewrite rules are handled since they're managed in a single place. It makes it easy to see everything that's going on at once.

Reply:
But those new to the repo won't know to look in this class on the off-chance there's some activation rules.